core/ppu.cpp

```diff
@@ -212,15 +214,21 @@ void PPU::HandleCpuWrite( u16 address, u8 data ) // NOLINT
                    and is set to bits 8-14 of _tempAddr
                    The _addrLatch is toggled
                  */
-                // TODO: Implement
+                _tempAddr.bit.coarseX = data & 0x1F; // shifts for high coarseX, nametablex/y
+                _tempAddr.bit.nametableX = ( data & 0x40 ) >> 6;
+                _tempAddr.bit.nametableY = ( data & 0x80 ) >> 7;
+                _addrLatch = true;
             } else {
                 /* Second Write
                    The entire data byte is the _tempAddr low byte
                    _tempAddr low byte is updated with the data byte
                   _tempAddr is copied to _vramAddr
                   _addrLatch is toggled
                  */
-                // TODO: Implement
+                _tempAddr.bit.coarseY = data & 0x1F;
+                _tempAddr.bit.fineY = ( data & 0xE0 ) >> 5;
+                _vramAddr.value = _tempAddr.value;
+                _addrLatch = false;
```

Comment:
As above, you should modify _tempAddr.value bits 0-7 directly. Since we've already written the top bits with the first write, we don't need to be as careful with masking.

1. mask _tempAddr.value with 0x7F00 to get the lower 8 bits
2. OR together the result from step 1 and data to set all bits.
3. Assign result to _tempAddr.value

```suggestion
                _tempAddr.value = ( _tempAddr.value & 0x7F00 ) | data;
                _vramAddr.value = _tempAddr.value;
                _addrLatch = false;
```